Network/Socket/Info.hsc

```diff
@@ -290,6 +304,15 @@ getAddrInfo hints node service = alloc getaddrinfo
     filteredHints = hints
 #endif
 
+getAddrInfoNE
+    :: Maybe AddrInfo
+    -> Maybe HostName
+    -> Maybe ServiceName
+    -> IO (NE.NonEmpty AddrInfo)
+getAddrInfoNE hints node service =
+    -- getAddrInfo never returns an empty list.
+    NE.fromList <$> getAddrInfo hints node service
```

Comment:
Since `getAddrInfoList` also does a non-empty check and throws an IO error if the list is empty, you could make `getAddrInfoNE` the "real" function and implement `getAddrInfoList` by fmapping `toList` over the result of `getAddrInfoNE`?

Reply:
Done.